providers/curation/github.js

```diff
@@ -37,42 +38,66 @@ class GitHubCurationService {
   }
 
   /**
-   * Process the fact that the given PR has been opened.
-   * @param {*} pr GitHub PR event object
+   * Enumerate all contributions in GitHub and in the store and updates any out of sync
    * @returns Promise indicating the operation is complete. The value of the resolved promise is undefined.
-   * @throws Exception with `code` === 404 if the given PR is missing. Other exceptions may be thrown related
-   * to interaction with GitHub or PR storage
    */
-  prOpened(pr) {
-    return this._storeAndValidateContribution(pr)
+  async syncAllContributions(client) {
+    let response = await client.pullRequests.getAll({
+      owner: this.options.owner,
+      repo: this.options.repo,
+      per_page: 100,
+      state: 'all'
+    })
+    this._processContributions(response.data)
+    while (this.github.hasNextPage(response)) {
+      response = await this.github.getNextPage(response)
+      this._processContributions(response.data)
+    }
+  }
+
+  async _processContributions(prs) {
+    for (let pr of prs) {
+      const storedContribution = await this.store.getContribution(pr.number)
+      if (new Date(get(storedContribution, 'pr.updated_at')).getTime() >= new Date(pr.updated_at).getTime()) continue
+      await this.updateContribution(pr)
+    }
   }
 
   /**
-   * Process the fact that the given PR has been closed.
-   * @param {*} pr GitHub PR event object
+   * Persist the updated contribution in the store and handle newly merged contributions
+   * @param {*} pr - The GitHub PR object
    * @returns Promise indicating the operation is complete. The value of the resolved promise is undefined.
-   * @throws Exception with `code` === 404 if the given PR is missing. Other exceptions may be thrown related
-   * to interaction with GitHub or PR storage
    */
-  prClosed(pr) {
-    return this.store.updateContribution(pr)
+  async updateContribution(pr) {
+    const curations = await this._getContributedCurations(pr.number, pr.head.sha)
+    const data = {
+      number: pr.number,
+      id: pr.id,
+      state: pr.state,
+      title: pr.title,
+      user: { login: get(pr, 'user.login') },
```

Comment:
is this going to be the logged in user the clearlydefinedbot user?

Reply:
this is who opened the PR, so as long as they are logged in it will be the contributor, bot the bot.

For example, https://github.com/clearlydefined/curated-data/pull/785

![image](https://user-images.githubusercontent.com/5168796/49304049-26dfea00-f480-11e8-8836-b0104bf12a00.png)